src/gates/parse_gates.cpp

```diff
@@ -47,19 +47,28 @@ Line split_instruction_and_args(std::string_view gate_str)
 {
     if(!lstk::contains(gate_str, ' '))
         return {gate_str, {}};
-
-    auto instr_split = lstk::split_on(gate_str,' ');
-    auto instruction = instr_split.at(0);
-    auto args = lstk::split_on(instr_split.at(1), ',');
-    auto semicolon_split = lstk::split_on(gate_str,';');
+    
+    auto semicolon_split = lstk::split_on(gate_str, ';');
+    gate_str = semicolon_split.at(0);
     auto annotation_line = semicolon_split.at(1);
-
+    
+    auto first_space = gate_str.find(' ');
+    auto instruction = gate_str.substr(0, first_space);
+    auto details = gate_str.substr(first_space + 1, gate_str.size());
```

Comment:
1) The problem is that we need to split on just the first space. If we have e.g.
```qasm
cx q[1],  q[2];
```
then `split_on` would return `{"cx", "q1,", "", "q[2];"}`. We need to split on the first space, and then split the remainder on ','.

2) Currently, there is a check for instructions without spaces at the top of `split_instruction_and_args`, so this never actually happens. But yes, I'll make this clearer and more robust.

Perhaps, the best way to address both these issues is to add a `split_on_first` function.

Reply:
Yes that would probably be best

Reply:
I would even be in favour of using a lightweight QASM parsing library like this https://github.com/softwareQinc/qasmtools

Though we've got to make sure it can handle the comment annotations